drivers/i3c/i3c_stm32.c

```diff
@@ -0,0 +1,2163 @@
+					  LL_I3C_DMA_GetRegAddr(i3c, LL_I3C_DMA_REG_STATUS), 0);
+	if (err != 0) {
+		return err;
+	}
+
+	/*Configure DMA TX */
+	err = i3c_stm32_dma_stream_config(
+		dev, &data->dma_tx, 0,
+		LL_I3C_DMA_GetRegAddr(i3c, LL_I3C_DMA_REG_DATA_TRANSMIT_BYTE));
+	if (err != 0) {
+		return err;
+	}
+
+	/*Configure DMA TC */
+	err = i3c_stm32_dma_stream_config(dev, &data->dma_tc, 0,
+					  LL_I3C_DMA_GetRegAddr(i3c, LL_I3C_DMA_REG_CONTROL));
+	if (err != 0) {
+		return err;
+	}
+
+	return err;
+}
+#endif
+
+static void i3c_stm32_controller_init(const struct device *dev)
+{
+	struct i3c_stm32_data *data = dev->data;
+	const struct i3c_stm32_config *config = dev->config;
+	I3C_TypeDef *i3c = config->i3c;
+
+#ifdef CONFIG_I3C_USE_IBI
+	LL_I3C_EnableHJAck(i3c);
+#endif
+
+	/* Configure FIFO */
+	LL_I3C_SetRxFIFOThreshold(i3c, LL_I3C_RXFIFO_THRESHOLD_1_4);
+	LL_I3C_SetTxFIFOThreshold(i3c, LL_I3C_TXFIFO_THRESHOLD_1_4);
+	LL_I3C_EnableControlFIFO(i3c);
+	LL_I3C_EnableStatusFIFO(i3c);
+
+	/* I3C Initialization */
+	LL_I3C_SetMode(i3c, LL_I3C_MODE_CONTROLLER);
+	LL_I3C_SetStallTime(i3c, 0x00);
+	LL_I3C_DisableStallACK(i3c);
+	LL_I3C_DisableStallParityCCC(i3c);
+	LL_I3C_DisableStallParityData(i3c);
+	LL_I3C_DisableStallTbit(i3c);
+	LL_I3C_DisableHighKeeperSDA(i3c);
+
+	LL_I3C_Enable(i3c);
+
+	LL_I3C_EnableIT_FC(i3c);
+	LL_I3C_EnableIT_CFNF(i3c);
+	LL_I3C_EnableIT_SFNE(i3c);
+	LL_I3C_EnableIT_RXFNE(i3c);
+	LL_I3C_EnableIT_TXFNF(i3c);
+	LL_I3C_EnableIT_ERR(i3c);
+	LL_I3C_EnableIT_WKP(i3c);
+
+#ifdef CONFIG_I3C_USE_IBI
+	LL_I3C_EnableIT_IBI(i3c);
+	LL_I3C_EnableIT_HJ(i3c);
+#endif
+
+	/* Bus will be idle initially */
+	data->msg_state = STM32_I3C_MSG_IDLE;
+	data->sf_state = STM32_I3C_SF_IDLE;
+	data->target_id = 0;
+#ifdef CONFIG_I3C_USE_IBI
+	data->ibi_payload = 0;
+	data->ibi_payload_size = 0;
+	data->ibi_target_addr = 0;
+#endif
+}
+
+/* Initializes the I3C device and I3C bus */
+static int i3c_stm32_init(const struct device *dev)
+{
+	const struct i3c_stm32_config *config = dev->config;
+	struct i3c_stm32_data *data = dev->data;
+	int ret;
+
+#ifdef CONFIG_I3C_STM32_DMA
+	ret = i3c_stm32_init_dma(dev);
+
+	if (ret != 0) {
+		LOG_ERR("Failed to init I3C DMA, err=%d", ret);
+		return ret;
+	}
+#endif
+
+	k_sem_init(&data->device_sync_sem, 0, K_SEM_MAX_LIMIT);
+
+	/* initialize mutex used when multiple transfers
+	 * are taking place to guarantee that each one is
+	 * atomic and has exclusive access to the I3C bus.
+	 */
+	k_mutex_init(&data->bus_mutex);
+
+	/* initialize semaphore used when multiple ibi requests are taking place */
+#ifdef CONFIG_I3C_USE_IBI
+	k_sem_init(&data->ibi_lock_sem, 1, 1);
+#endif
+	ret = i3c_addr_slots_init(dev);
+	if (ret != 0) {
+		LOG_ERR("Addr slots init fail, err=%d", ret);
+		return ret;
+	}
+
+	config->irq_config_func(dev);
+	i3c_stm32_configure(dev, I3C_CONFIG_CONTROLLER, NULL);
+	i3c_stm32_controller_init(dev);
+
+#ifdef CONFIG_PM_DEVICE_RUNTIME
+	(void)pm_device_runtime_enable(dev);
+#endif
```

Comment:
This should not be here, this would give the user no way to disable this individually.
If this is to be called, then there is to be `zephyr,pm-device-runtime-auto` within the dts node for this and will be called right after the init function